src/gui/rss/articlelistwidget.cpp

```diff
@@ -55,28 +55,38 @@ void ArticleListWidget::setRSSItem(RSS::Item *rssItem, bool unreadOnly)
 {
     // Clear the list first
     clear();
+    m_rssArticleToListItemMapping.clear();
     if (m_rssItem)
         m_rssItem->disconnect(this);
 
     m_unreadOnly = unreadOnly;
     m_rssItem = rssItem;
-    if (!m_rssItem) return;
-
-    m_rssItem = rssItem;
-    connect(m_rssItem, &RSS::Item::newArticle, this, &ArticleListWidget::handleArticleAdded);
-    connect(m_rssItem, &RSS::Item::articleRead, this, &ArticleListWidget::handleArticleRead);
-    connect(m_rssItem, &RSS::Item::articleAboutToBeRemoved, this, &ArticleListWidget::handleArticleAboutToBeRemoved);
-
-    foreach (auto article, rssItem->articles()) {
-        if (!(m_unreadOnly && article->isRead()))
-            addItem(createItem(article));
+    if (m_rssItem) {
```

Comment:
You don't like such long `if`s, do you? If I were you, maybe the following could make me happier:
```c++
template <typename Func, bool enable>
class Checker
{
public:
    Checker(Func f, bool checkOnEntry, bool checkOnExit)
       : f_{f}
       , checkOnExit_(checkOnExit) {
              if (checkOnEntry) {
                  f_();
              }
       }
       ~Checker() {
              if (checkOnExit_) {
                  f_();
              }
       }
private:
    Func f_;
    bool checkOnExit_;
};

template <typename F>
class Checker<F,false> {
    public:
        Checker(F, bool, bool) {
        }
};

template <bool v, typename Func>
Checker<Func, v> makeChecker(Func f, bool onEntry, bool onExit) {
     return Checker<Func, v>(f, onEntry, onExit);
}
```
and then:
```c++
#ifndef NDEBUG
static constexpr bool doChecks = false;
#else
static constexpr bool doChecks = true;
#endif

void A::f() {
    auto checker = makeChecker<doChecks>([this](){this->checkInvariant();}, true, true);
    if (cond) 
       return;
   doWork(0;
}
```